addons/sale_order_dates/models/sale_order.py

```diff
@@ -24,30 +24,28 @@ class SaleOrder(models.Model):
                                           "processed as soon as possible. In that case the expected "
                                           "date will be computed using the default method: based on "
                                           "the Product Lead Times and the Company's Security Delay.")
-    effective_date = fields.Date(compute='_compute_picking_ids', string='Effective Date', store=True,
-                                 help="Date on which the first Delivery Order was created.")
+    effective_date = fields.Date(compute='_compute_effective_date', string='Effective Date', store=True,
+                                 help="Date on which the first shipment was successfully delivered to customer.")
 
-    @api.depends('date_order', 'order_line.customer_lead')
+    @api.depends('order_line.customer_lead', 'confirmation_date', 'picking_policy')
     def _compute_commitment_date(self):
         """Compute the commitment date"""
         for order in self:
             dates_list = []
-            order_datetime = fields.Datetime.from_string(order.date_order)
+            confirm_date = fields.Datetime.from_string(order.confirmation_date or fields.Datetime.now())
             for line in order.order_line.filtered(lambda x: x.state != 'cancel'):
-                dt = order_datetime + timedelta(days=line.customer_lead or 0.0)
+                dt = confirm_date + timedelta(days=line.customer_lead or 0.0)
                 dates_list.append(dt)
             if dates_list:
                 commit_date = min(dates_list) if order.picking_policy == 'direct' else max(dates_list)
                 order.commitment_date = fields.Datetime.to_string(commit_date)
 
-    def _compute_picking_ids(self):
-        super(SaleOrder, self)._compute_picking_ids()
+    @api.depends('order_line', 'order_line.qty_delivered')
```

Comment:
Triggers seems a bit unrelated to the fields used in the method. Are you sure everything is present ?

Reply:
We can put here trigger "picking_ids.date_done" to call when shipment goes into done state and 
it will min(done_date) of done pickings.